modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java

```diff
@@ -141,14 +141,18 @@ public ListTypeEntry updateListTypeEntry(
 		ListTypeEntry listTypeEntry = listTypeEntryPersistence.findByPrimaryKey(
 			listTypeEntryId);
 
-		_validateExternalReferenceCode(
-			externalReferenceCode, listTypeEntry.getCompanyId(),
-			listTypeEntry.getListTypeDefinitionId(), listTypeEntryId);
+		if (Validator.isNotNull(externalReferenceCode)) {
+			_validateExternalReferenceCode(
+				externalReferenceCode, listTypeEntry.getCompanyId(),
+				listTypeEntry.getListTypeDefinitionId(), listTypeEntryId);
 
-		_validateName(nameMap);
+			listTypeEntry.setExternalReferenceCode(externalReferenceCode);
+		}
 
-		listTypeEntry.setExternalReferenceCode(externalReferenceCode);
-		listTypeEntry.setNameMap(nameMap);
+		if (!nameMap.isEmpty()) {
```

Comment:
Hey @mateussgn , I think that with this change it can cause a NPE when `nameMap` is null